packages/esm-patient-conditions-app/src/conditions/conditions-form.component.tsx

```diff
@@ -1,479 +1,232 @@
+import RadioButtonGroup from "carbon-components-react/es/components/RadioButtonGroup";
+import Search from "carbon-components-react/es/components/Search";
+import SearchSkeleton from "carbon-components-react/es/components/Search/Search.Skeleton";
+import { Tile } from "carbon-components-react/es/components/Tile";
+import {
+  searchConditionConcepts,
   createPatientCondition,
-  updatePatientCondition,
+  CodedCondition,
 } from "./conditions.resource";
-import { DataCaptureComponentProps } from "../types";
+import styles from "./conditions-form.scss";
 
-type ConditionsFormProps = DataCaptureComponentProps & {
-  match: any;
+interface ConditionsFormProps {
   patientUuid: string;
-};
-
-interface Condition {
-  conditionUuid?: string;
-  inactivityDate?: string;
-  conditionName: string;
-  clinicalStatus: string;
-  onsetDateTime: string;
 }
-
-const ConditionsForm: React.FC<ConditionsFormProps> = ({
-  match,
-  patientUuid,
-  entryStarted = () => {},
-  entryCancelled = () => {},
-  closeComponent = () => {},
-}) => {
-  const formRef = useRef<HTMLFormElement>(null);
-  const [conditionName, setConditionName] = useState("");
-  const [conditionUuid, setConditionUuid] = useState("");
-  const [clinicalStatus, setClinicalStatus] = useState(null);
-  const [onsetDateTime, setOnsetDateTime] = useState(null);
-  const [enableCreateFormButtons, setEnableCreateFormButtons] = useState(false);
-  const [enableEditFormButtons, setEnableEditFormButtons] = useState(true);
-  const [viewEditForm, setViewEditForm] = useState(true);
-  const [inactiveStatus, setInactiveStatus] = useState(false);
-  const [inactivityDate, setInactivityDate] = useState("");
-  const [formChanged, setFormChanged] = useState(false);
+const ConditionsForm: React.FC<ConditionsFormProps> = ({ patientUuid }) => {
   const { t } = useTranslation();
-  const history = useHistory();
-
-  useEffect(() => {
-    if (conditionName && onsetDateTime && clinicalStatus) {
-      setEnableCreateFormButtons(true);
-    } else {
-      setEnableCreateFormButtons(false);
-    }
-  }, [conditionName, onsetDateTime, clinicalStatus]);
-
-  useEffect(() => {
-    if (viewEditForm && formChanged) {
-      setEnableEditFormButtons(true);
-    } else {
-      setEnableEditFormButtons(false);
-    }
-  }, [viewEditForm, formChanged]);
+  const searchTimeoutInMs = 300;
+  const session = useSessionUser();
+  const [onsetDate, setOnsetDate] = React.useState(null);
+  const [searchTerm, setSearchTerm] = React.useState("");
+  const [searchResults, setSearchResults] = React.useState<
+    Array<CodedCondition>
+  >(null);
+  const [condition, setCondition] = React.useState<any>(null);
+  const [clinicalStatus, setClinicalStatus] = React.useState("active");
+  const [endDate, setEndDate] = React.useState(null);
+  const [isSearching, setIsSearching] = React.useState(false);
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
```

Comment:
`searchTerm`, `searchResults`, `condition`, `isSearching` and `isSubmitting` are connected states (e.g. when condition is set we don't care about the searchTerm or searchResults) and `handleConditionChange` triggers 3 state changes at once.

It might be a good idea to combine the states into a singe state, which would make handling easier and guarantee state consistency. If you need some inspiration on how to combine them, reach out to me and we can have a look together.

Reply:
Sure - I'd like to hear more about this approach @nmalyschkin